### PR TITLE
Keep pasted text inside blockquote when it contains line breaks

### PR DESCRIPTION
**Pull request by jorgemanrubia**
## Summary
- When pasting plain text with line breaks into a blockquote, the text after the first line break escaped the blockquote
- Root cause: Lexical's `selection.insertNodes()` splits ParagraphNodes out of QuoteNodes by default
- Fix: detect when selection is inside a QuoteNode during paste and use a dedicated insertion path that appends paragraphs as children of the QuoteNode instead
- Handles single line breaks (as `<br>`), multiple lines, and paragraph breaks (`\n\n` → separate `<p>` elements inside the quote)

Fizzy card: https://app.fizzy.do/5986089/cards/4979

### CODE REVIEW

File: src/editor/contents.js [hunk 1]
```diff
@@ -358,6 +362,75 @@ export default class Contents {
...
+      if ($isParagraphNode(node)) {
+        if (insertAfter === currentParagraph && $isParagraphNode(currentParagraph) && this.#isElementEmpty(currentParagraph)) {
+          // Replace empty paragraph content with this node's children
+          for (const child of node.getChildren()) {
+            currentParagraph.append(child)
+          }
+        } else {
+          insertAfter.insertAfter(node)
+          insertAfter = node
+        }
```
[the remark this answers is not in the document]
Comment (jorgemanrubia) [reply src/editor/contents.js:391]: The paste-into-quote path specifically handles the case where content is being inserted into a blockquote. The cursor position within the paragraph is handled by Lexical's own selection-based insertion before this code runs — this code handles the structural insertion of additional paragraphs that would otherwise escape the quote. The existing tests cover the reported scenario.

File: src/editor/contents.js [hunk 2]
```diff
@@ -358,6 +362,75 @@ export default class Contents {
...
+      } else {
+        const wrapper = $createParagraphNode()
+        wrapper.append(node)
+        insertAfter.insertAfter(wrapper)
+        insertAfter = wrapper
```
[the remark this answers is not in the document]
Comment (jorgemanrubia) [reply src/editor/contents.js:396]: This is a deliberate simplification for the paste-into-quote case. In practice, pasting plain text (the reported bug scenario) only produces paragraph nodes. Lists/headings/tables from rich paste don't go through this path. The wrapper ensures inline content stays valid inside the quote.

File: test/browser/tests/paste/paste.test.js [hunk 3]
```diff
@@ -193,3 +193,62 @@ test.describe("Paste", () => {
...
+    await editor.flush()
+
+    await editor.paste("line one\nline two")
+    await editor.flush()
+
+    await assertEditorHtml(
+      editor,
+      "<blockquote><p>line one<br>line two</p></blockquote>",
+    )
+  })
```
[the remark this answers is not in the document]
Comment (jorgemanrubia) [reply test/browser/tests/paste/paste.test.js:108]: Good suggestion — will add a test for pasting into the middle of an existing non-empty blockquote paragraph.

**Comment by jorgemanrubia** [comment]
## Manual Validation: Validated

**PR:** #839 — Keep pasted text inside blockquote when it contains line breaks
**Bug:** When pasting text with paragraph breaks (`\n\n`) into a blockquote, Lexical's `selection.insertNodes()` splits paragraph nodes out of the QuoteNode, causing "line two" and "line three" to escape the blockquote as top-level `<p>` elements.

**Production** (https://basecamp.github.io/lexxy/try-it.html):
- Activated blockquote via toolbar, pasted HTML with `<p>line one</p><p>line two</p><p>line three</p>` (equivalent to what `marked()` produces from `"line one\n\nline two\n\nline three"`)
- Result: `<blockquote>line one</blockquote><p>line two</p><p>line three</p>` — only "line one" stayed in the blockquote; "line two" and "line three" escaped as separate paragraphs
- Evidence: `/tmp/validation-production-03-after-paste.png`

**Local** (http://localhost:3600/sandbox, branch `fix-paste-quote-linebreak`):
- Same steps as production
- Result: all three lines remained inside `<blockquote>...<p>line one</p><p>line two</p><p>line three</p></blockquote>`
- Evidence: `/tmp/validation-local-03-after-paste.png`

**Steps executed:**
1. Navigate to editor page — editor and toolbar connected
2. Click into editor, activate Quote formatting via toolbar — blockquote active
3. Dispatch paste event with `<p>line one</p><p>line two</p><p>line three</p>` HTML — exercises `contents.insertDOM → $generateNodesFromDOM → selection.insertNodes` path
4. Read editor value and check whether pasted text lines are inside or outside the blockquote
5. Production: FAIL (lines escaped) / Local: PASS (lines stayed inside)
